memo_app.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'json'
+require 'securerandom'
+
+MEMOS_PATH = File.join(settings.public_folder, 'memos.json')
+
+helpers do
+  def h(text)
+    Rack::Utils.escape_html(text)
+  end
+end
+
+def load_memos
+  JSON.parse(File.read(MEMOS_PATH))
+end
+
+def save_memos(memos)
+  File.open(MEMOS_PATH, 'w') do |file|
+    file.write(JSON.generate(memos))
+  end
+end
+
+def find_memo(memos, id)
+  memos.find { |memo| memo['id'] == id }
+end
+
+get '/' do
+  save_memos([]) unless File.exist?(MEMOS_PATH)
+
+  @memos = load_memos
+  erb :index
+end
+
+get '/memos/new' do
+  erb :new
+end
+
+post '/memos' do
+  memo = {
+    id: SecureRandom.uuid,
+    title: params[:title],
+    content: params[:content]
+  }
+
+  memos = load_memos
+  memos << memo
+
+  save_memos(memos)
+
+  redirect '/'
+end
+
+get '/memos/:id' do
+  memos = load_memos
+  @memo = find_memo(memos, params[:id])
+
+  if @memo
+    erb :memo
+  else
+    status 404
+    erb :not_found
```

Comment:
`not_found` メソッドの説明の通り、数値の404を返すだけで `not_found` メソッドが実行されるので、ここであらためて `erb :not_found` を指定する必要はないかなと思います。

```suggestion
    404
```

Reply:
以下のコードで動作確認を行いました。
おっしゃるとおり、ステータスも404を返して、Webブラウザにも「Not Found」と表示されました。
以下のコードで、十分に機能を果たしていると思います。
```
not_found do
end
```

Reply:
僕が書いたのは、ここでは `404` を返すとsinataraの仕組みで `not_found` が実行されるので、 `erb` の設定などはそちらにまとめられますよ、ということでした。このままだとここで `erb` も設定しているので、あまり `not_found` メソッドの意味がないです。

```ruby
get '/memos/:id' do
  memos = load_memos
  @memo = find_memo(memos, params[:id])

  if @memo
    erb :memo
  else
    404
  end
end

# 404のときはこちらが読み込まれる
not_found do
  erb :not_found
end
```